app/__init__.py

```diff
@@ -29,7 +40,57 @@
 
 from app import routes, template_helpers
 
+#caching all the dataone original sources in memory (only update the datasources after 120 secs)
+@cache.cached(timeout=120, key_prefix='cache_original sources for dataone')
+def get_original_dataone_sources():
+    datasources = dict()
+    # XML SCRAPING
+    mydoc = minidom.parse(urlopen("https://cn.dataone.org/cn/v2/node/"))
+    tree = ET.parse(urlopen("https://cn.dataone.org/cn/v2/node/"))
+
+
+    #node = mydoc.getElementsByTagName('node')
+    root = tree.getroot()
+    print(len(root))
```

Comment:
looks like you left a debug statement in here - if it's handy, you could make it a debug log, but if you don't need it, please remove it